packages/atomic/src/components/commerce/product-template-components/atomic-product-text/e2e/atomic-product-text.e2e.ts

```diff
@@ -0,0 +1,94 @@
+import {test, expect} from './fixture';
+
+test.describe('default', async () => {
+  test.beforeEach(async ({productText}) => {
+    await productText.load();
+    await productText.hydrated.first().waitFor();
+  });
+
+  test('should be accessible', async ({makeAxeBuilder}) => {
+    const accessibilityResults = await makeAxeBuilder().analyze();
+    expect(accessibilityResults.violations.length).toEqual(0);
+  });
+
+  test.describe('when field has no value and default is set', async () => {
+    test('should render default text', async ({productText}) => {
+      await productText.load({
+        args: {field: 'nonexistentField', default: 'Default Text'},
+      });
+      await productText.hydrated.first().waitFor();
+
+      expect(productText.textContent.first()).toContainText('Default Text');
+    });
+  });
+});
+
+test.describe('when using a field that supports highlights', async () => {
+  const fields = ['excerpt', 'ec_name'];
+
+  fields.forEach((field) => {
+    test.describe(`when displaying the ${field}`, async () => {
+      test.beforeEach(async ({productText}) => {
+        await productText.load({args: {field}});
+        await productText.hydrated.first().waitFor();
+      });
+
+      test(`should highlight the keywords in the ${field}`, async ({
+        productText,
+      }) => {
+        const keywordPattern = /^kayak/i;
+
+        const highlightedText =
+          await productText.highlightedText.allTextContents();
+
+        highlightedText.forEach((text) => {
+          expect(text).toMatch(keywordPattern);
+        });
+      });
+    });
+
+    test(`should not highlight the keywords in the ${field} when shouldHighlight is false`, async ({
+      productText,
+    }) => {
+      await productText.load({
+        args: {field: 'excerpt', shouldHighlight: false},
+      });
+      await productText.hydrated.first().waitFor();
```

Comment:
can you remove that beforeEach and put it in the `should highlight the keywords in the ${field}` since it is being overwritten in the next test anyway ?